boards/BMS/Test/Src/state_machine_test.cpp

```diff
@@ -62,6 +57,16 @@ class BmsStateMachineTest : public testing::Test
             App_SharedStateMachine_GetCurrentState(state_machine));
     }
 
+    std::vector<const struct State *> GetAllStates(void)
```

Comment:
Came up with this helper function for us to test something in all states. I felt that if we were to test whether a CAN message is sent correctly in all 4 states, splitting that across 4 `TEST_F` would be quite redundant.